content/blog/2023/2023-08-30-fairness-equal-odds-postprocessing.md

```diff
@@ -0,0 +1,57 @@
++++
+author = "Žan Mervič"
+date = "2023-08-30"
+draft = false
+title = "Orange Fairness - Equal Odds Postprocessing"
+type = "blog"
+thumbImage = "/blog_img/2023/2023-08-30-fairness-equal-odds-postprocessing.png"
+frontPageImage = "/blog_img/2023/2023-08-30-fairness-equal-odds-postprocessing.png"
+blog = ["fairness", "equal odds postprocessing"]
+shortExcerpt = "In this blog, we delve into the Equal Odds Postprocessing widget, a tool designed to enhance fairness in machine learning models. We break down how the algorithm works by modifying predictions to meet Equalized Odds criteria. Using a real-world example with the German credit dataset, we demonstrate its efficacy in improving fairness metrics while marginally affecting accuracy."
+longExcerpt = "Explore the Equal Odds Postprocessing widget in Orange, designed to fine-tune your model's fairness. We explain how the algorithm operates and showcase its effectiveness with an example using the German credit dataset."
++++
+
+
+In the [previous blog post]({{< ref "2023-08-28-fairness-adversarial-debiasing.md" >}}), we discussed the Adversarial Debiasing model, a bias-aware model. This blog post will discuss the Equal Odds Postprocessing widget, a bias-aware post-processor, which can be used with any model to mitigate bias in its predictions.
+
+### Equal Odds Postprocessing:
+
+The [Equal Odds Postprocessing](https://arxiv.org/abs/1610.02413) widget is a post-processing type of fairness mitigation algorithm designed to ensure fairness in supervised learning. It modifies the predictions of any given classifier to meet certain fairness criteria, specifically focusing on "Equalized Odds" or more relaxed criteria like Equal Opportunity. Because it is a post-processing algorithm, it does not require any support from the model (like reweighing, which requires the model to support weights) or changes to the model (like adversarial debiasing, which requires the model to be a neural network). This makes it a versatile algorithm that can be used with any model.
+
+The widget Equalized Odds Postprocessing represents a novelty in the Orange environment, as until now, there has been no widget with similar functionality. Its operation is similar to other Orange widgets representing models, except it also expects one of the Orange models as an input. It operates by first learning from the predictions of the model's training data. Then, it applies the Equalized Odds Postprocessing algorithm from the [AIF360](https://aif360.res.ibm.com/) library to the predictions the selected model makes on the test data. This ensures that the adjusted predictions conform to the fairness definition of equalized odds.
```

Comment:
represents a novelty - too ornamental, also, if something is novel, then "as until now, there has been no widget with similar functionality" is a tautology and thus needless.

"Its operation is similar to other Orange widgets representing models, except it also expects one of the Orange models as an input." -> "It needs a model as an input." (also, does it need a learner or model?)

"It operates by first learning from the predictions of the model's training data." - I have not idea what you wanted to say with this

"This ensures that the adjusted predictions conform to the fairness definition of equalized odds." -> This is already said with the algorithm name, why repeat it here?